src/include/setround.h

```diff
@@ -0,0 +1,29 @@
+#ifndef __SETROUND_H__
+#define __SETROUND_H__
+#pragma STDC FENV_ACCESS ON
+#include <fenv.h>
+#include <stdint.h>
+#include "x86emu.h"
+#include "emu/x86emu_private.h"
+// set the rounding mode to the emulator's one, and return the old one
+static inline int setround(x86emu_t* emu) {
+    int ret = fegetround();
+    int rounding_direction;
+    switch (emu->cw.f.C87_RD) {
+        case ROUND_Nearest:
+            rounding_direction = FE_TONEAREST;
+            break;
+        case ROUND_Down:
+            rounding_direction = FE_DOWNWARD;
+            break;
+        case ROUND_Up:
+            rounding_direction = FE_UPWARD;
+            break;
+        case ROUND_Chop:
+            rounding_direction = FE_TOWARDZERO;
+            break;
+    }
+    fesetround(rounding_direction);
+    return ret;
+}
```

Comment:
Should I name this function `fpu_setround` and move it to `src/x87emu_private.{h,c}`? I also make a patch to wrapper some functions of libm to set their rounding mode, which uses this function. (To avoid conflicts, I will submit it after this PR is merged.)

Reply:
Yeah, renaming it would be better (and consistant with box64 also)

Reply:
I just found it is probably not good to move that to `src/emu/x87emu_private.h`, for `#pragma STDC FENV_ACCESS ON` may disable some optimization about float. How about move it to `src/emu/x87emu_setround.h`?

Reply:
ok. Keep it in it's file, just rename to `fpu_setround(...)`.